lib/services/elasticsearch.js

```diff
@@ -153,718 +168,635 @@ class ElasticSearch extends Service {
-              deletedAt: Date.now(),
-              updater: this._getUserId(request)
-            }
+            this.throw('index_already_exists', index, internalMessage);
           }
-        };
-
-        return this.client.update(esRequest)
-          .then(r => this.refreshIndexIfNeeded(esRequest, r))
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
+        }
       })
-      .catch(error => Bluebird.reject(
-        this.esWrapper.formatESError(error))
-      );
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
-   * Delete all documents matching the provided filters
+   * Creates an empty collection. Mapping will be applied if supplied.
    *
-   * @param {Request} request
-   * @returns {Promise} resolve an object with ids
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Object} mappings - Collection mappings in ES format
+   *
+   * @returns {Promise.<Object>} { }
    */
-  deleteByQuery(request) {
-    const
-      esRequestSearch = initESRequest(request, ['from', 'size', 'scroll']),
-      esRequestBulk = initESRequest(request, ['refresh']),
-      ts = Date.now();
-
-    assertWellFormedRefresh(esRequestBulk);
-
-    esRequestSearch.body = request.input.body;
-    esRequestSearch.scroll = '30s';
-    esRequestBulk.body = [];
-
-    if (!esRequestSearch.body.query ||
-      !(esRequestSearch.body.query instanceof Object)) {
-      return Bluebird.reject(this.getError('empty_query'));
-    }
-
-    return getAllIdsFromQuery(this.client, addActiveFilter(esRequestSearch))
-      .then(ids => {
-        ids.forEach(id => {
-          esRequestBulk.body.push(
-            {update:
-              {_index: esRequestBulk.index,
-                _type: esRequestBulk.type,
-                _id: id}
-            });
-          esRequestBulk.body.push(
-            {doc:
-              {_kuzzle_info:
-                { active: false,
-                  deletedAt: ts,
-                  updater: this._getUserId(request)}}
-            });
-        });
+  createCollection (index, collection, mappings = {}) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: { mappings: _.merge(mappings, this.config.commonMapping) }
+    };
 
-        if (esRequestBulk.body.length === 0) {
-          return Bluebird.resolve({ids: []});
+    return this.collectionExists(index, collection)
+      .then(exists => {
+        if (exists) {
+          return this.updateMapping(index, collection, mappings);
         }
 
-        return this.client.bulk(esRequestBulk)
-          .then(() => this.refreshIndexIfNeeded(esRequestBulk, {ids}))
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
+        this._checkMappings(esRequest.body.mappings);
+
+        return this.client.indices.create(esRequest)
+          .then(() => {})
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this function is empty or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1186)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AEmptyBlock)